File: src/chalk.Server/DTOs/Responses/UserResponse.cs
using System.Text.Json.Serialization;

namespace chalk.Server.DTOs.Responses;

[Serializable]
[method: JsonConstructor]
public record UserResponse(
    [property: JsonRequired]
    [property: JsonPropertyName("id")]
    long Id,
    [property: JsonRequired]
    [property: JsonPropertyName("email")]
    string Email,
    [property: JsonRequired]
    [property: JsonPropertyName("firstName")]
    string FirstName,
    [property: JsonRequired]
    [property: JsonPropertyName("lastName")]
    string LastName,
    [property: JsonRequired]
    [property: JsonPropertyName("displayName")]
    string DisplayName,
    [property: JsonRequired]
    [property: JsonPropertyName("description")]
    string? Description,
    [property: JsonRequired]
    [property: JsonPropertyName("imageUrl")]
    string? ImageUrl,
    [property: JsonRequired]
    [property: JsonPropertyName("createdDate")]
    string CreatedDate,
    [property: JsonRequired]
    [property: JsonPropertyName("directMessages")]
    IEnumerable<ChannelDTO> DirectMessages,
    [property: JsonRequired]
    [property: JsonPropertyName("courses")]
    IEnumerable<CourseDTO> Courses,
    [property: JsonRequired]
    [property: JsonPropertyName("organizations")]
    IEnumerable<OrganizationDTO> Organizations
);

File: src/Web/src/lib/api/responses/UserResponse.ts
type UserResponse = {
  id: number;
  email: string;
  firstName: string;
  lastName: string;
  displayName: string;
  description: string | null;
  imageUrl: string | null;
  createdOnUtc: string;
  updatedOnUtc: string;
  directMessages: PartialChannelResponse[];
  courses: PartialCourseResponse[];
  organizations: PartialOrganizationResponse[];
};

export type PartialChannelResponse = {
  id: number;
};

export type PartialCourseResponse = {
  id: number;
  name: string;
  code: string | null;
};

export type PartialOrganizationResponse = {
  id: number;
  name: string;
};

export default UserResponse;
